pkg/client/userd/trafficmgr/install.go

```diff
@@ -291,14 +300,27 @@ func (ki *installer) EnsureAgent(c context.Context, obj k8sapi.Workload,
 	var svc k8sapi.Object
 	autoInstall, err := useAutoInstall(podTemplate)
 	if err != nil {
-		return "", "", err
+		return "", "", 0, err
+	}
+
+	matchingSvc, err := install.FindMatchingService(c, portNameOrNumber, svcName, namespace, podTemplate.Labels)
```

Comment:
The creation of a personal intercept was recently divided into two steps to get rid of some friction. One to ensure that an intercept is indeed possible (the `trafficmgr.CanIntercept()` which is executed before any questions are asked regarding ingress), and one that actually creates the intercept (the `trafficmgr.AddIntercept()`). The new actions performed here to find the matching service and its port, should probably be performed as part of the first step so that errors are caught early.

Reply:
that makes sense